### DISCUSSION TITLE
Evaluate JSON responses in scripting?

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#16267] [Q&A]
@gemu2015 When using rules, I can use `on webquery#Data$|failed do...`. Does a similar function exist in scripting? I see the >E section, so would it be something like this:
```
#define USE_WEBSEND_RESPONSE
```
```
http("ntfy.sh/topic_xxxx" "[Title: 32v Charger 1] %1curr%a while relay off") 
```
```
>E
rslt=Result

>S
string1="failed"
if ins(rslt string1) {
}
```
Also noticing that on MacOS 12.2.1, Script Editor is auto-replacing quotes with smart quotes. Not sure if that will cause an issue when transferring the script over.

**Comment by gemu2015** [comment]
yes in principle that should work for JSON responses.

> Also noticing that on MacOS 12.2.1, Script Editor is auto-replacing quotes with smart quotes. Not sure if that will cause an issue when transferring the script over.


i don't know what you mean, can you give an example ?

**Comment by pkkrusty** [comment]
Video attached

https://user-images.githubusercontent.com/79770016/185735844-a1882ce0-a74c-4f60-bbc3-263b6f2ab4d0.mp4

**Reply by gemu2015** [reply]
this is not related to the script editor but the default behavior of macOS.
go to keyboard setup and change the default quotes:
<img width="801" alt="Bildschirmfoto 2022-08-20 um 11 55 39" src="https://user-images.githubusercontent.com/11647075/185741145-ca152247-548e-4d03-bbd3-001c26008b7b.png">

**Comment by pkkrusty** [comment]
Having trouble making http("" "") command work.

```
=>webquery http://ntfy.sh/topic_xxxx POST [Title: 32v Charger 1] %2curr% while relay off
```
works fine and generates a JSON response that I can use, but 
```
res=http("ntfy.sh/topic_xxxx" "[Title: 32v Charger 1] %2curr% while relay off")
```
isn't generating anything, either from the POST itself, or in the >E section. Am I missing something?

**Comment by gemu2015** [comment]
res=http("ntfy.sh/topic_xxxx" "[Title: 32v Charger 1] %2curr% while relay off") is a script cmd where var replacement with %% does not work. and a slash is missing

try this:
str="/[Title: 32v Charger 1]+s(2curr)+"while relay off"
res=http("ntfy.sh/topic_xxxx" str)

**Reply by pkkrusty** [reply]
Still nothing when I try 
```
str="/[Title: 32v Charger 1]+s(2curr)+"while relay off"
res=http("ntfy.sh/topic_xxxx" str)
```
In the console I get a `var not found:` with no var listed

**Reply by gemu2015** [reply]
I missed a quote, sorry
str="/[Title: 32v Charger 1]"+s(2curr)+"while relay off"
and this string is longer then the default. you must define a larger string size
var not found:  did you check you defined all vars: str,res,curr
